src/lib/viewers/archive/ArchiveViewer.js

```diff
@@ -0,0 +1,310 @@
+        },
+    },
+    {
+        type: 'file',
+        absolute_path: 'test/level-1/level-2-2/コラボレーションフォルダマニュアル.pptx',
+        name: 'コラボレーションフォルダマニュアル.pptx',
+        modified_at: '19-Nov-13 15:33',
+        size: 2899183,
+        path_collection: {
+            total_count: 3,
+            entries: [
+                { type: 'folder', absolute_path: 'test/', name: 'test' },
+                {
+                    type: 'folder',
+                    absolute_path: 'test/level-1/',
+                    name: 'level-1',
+                },
+                {
+                    type: 'folder',
+                    absolute_path: 'test/level-1/level-2-2/',
+                    name: 'level-2-2',
+                },
+            ],
+        },
+        parent: 'level-2-2',
+        item_collection: null,
+    },
+    {
+        type: 'file',
+        absolute_path: 'test/level-1/level-2/content-explorer-safari-blinking.mov',
+        name: 'content-explorer-safari-blinking.mov',
+        modified_at: '19-Nov-13 16:25',
+        size: 4472460,
+        path_collection: {
+            total_count: 3,
+            entries: [
+                { type: 'folder', absolute_path: 'test/', name: 'test' },
+                {
+                    type: 'folder',
+                    absolute_path: 'test/level-1/',
+                    name: 'level-1',
+                },
+                {
+                    type: 'folder',
+                    absolute_path: 'test/level-1/level-2/',
+                    name: 'level-2',
+                },
+            ],
+        },
+        parent: 'level-2',
+        item_collection: null,
+    },
+];
+
+class ArchiveViewer extends BaseViewer {
+    /**
+     * @inheritdoc
+     */
+    setup() {
+        if (this.isSetup) {
+            return;
+        }
+
+        // Call super() first to set up common layout
+        super.setup();
+
+        this.archiveEl = this.createViewer(document.createElement('div'));
+        this.archiveEl.className = 'bp-archive';
+    }
+
+    /**
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        if (this.archiveComponent) {
+            this.archiveComponent.destroy();
+        }
+        super.destroy();
+    }
+
+    /**
+     * Loads a archive file.
+     *
+     * @return {void}
+     */
+    load() {
+        super.load();
+
+        return Promise.all([this.loadAssets(JS, CSS), this.getRepStatus().getPromise()])
+            .then(this.postLoad)
+            .catch(this.handleAssetError);
+    }
+
+    /**
+     * Loads archive file representation.
+     *
+     * @private
+     * @return {Promise} promise to get archive content
+     */
+    postLoad = () => {
+        const { representation } = this.options;
+        const template = representation.content.url_template;
+        const contentUrl = this.createContentUrlWithAuthParams(template);
+        this.startLoadTimer();
+
+        this.api.get(contentUrl).then(data => {
```

Comment:
This promise isn't being returned, which means errors within it won't be caught by the parent `.catch`. This promise chain also still needs to be flattened. Let's move the `this.api.get` call and the logic in its anonymous `.then` clause each to separate methods and call them via two separate `.then` clauses in the main Promise.all chain.